### PR TITLE
Dump logs to stderr when syslog write fails

### PR DESCRIPTION
**Pull request by byewokko**

### CODE REVIEW

File: asab/log.py [hunk 1]
```diff
@@ -446,13 +446,23 @@ def _on_write(self):
...
+		error_counter = 0
 		while not self._queue.empty():
 			msg = self._queue.get_nowait()
 			try:
 				self._socket.sendall(msg)
 			except Exception as e:
-				print("Error when writing to syslog '{}'".format(self._address), e, file=sys.stderr)
+				if error_counter > 100:
+					print("Critical error when writing to syslog. Exiting.", file=sys.stderr)
+					sys.exit(1)
```
Comment (byewokko) [inline asab/log.py:457]: it should not exit, use `break` instead

File: asab/log.py [hunk 2]
```diff
@@ -446,13 +446,23 @@ def _on_write(self):
 		self._write_ready = True
 		self._loop.remove_writer(self._socket)
 
+		error_counter = 0
```
Comment (ateska) [inline asab/log.py:449]: Add a comment to `on_write` -> this method is a contingency dump when the socket is not ready.

File: asab/log.py (hunk 1 above)
Comment (ateska) [inline asab/log.py:457]: No exit - as discussed on the review.

**Review by ateska — CHANGES_REQUESTED** [review]

File: asab/log.py [hunk 3]
```diff
@@ -446,13 +446,23 @@ def _on_write(self):
...
 		while not self._queue.empty():
 			msg = self._queue.get_nowait()
 			try:
 				self._socket.sendall(msg)
 			except Exception as e:
-				print("Error when writing to syslog '{}'".format(self._address), e, file=sys.stderr)
+				if error_counter > 100:
```
Comment (byewokko) [inline asab/log.py:455]: there should not be any counter and exit. simply print the error and the socket error will hopefully fix itself in a bit

Comment (byewokko) [reply asab/log.py:455]: also do not `self._enqueue(msg)`
the msg should be thrown away

**Comment by ateska** [comment]
Don't exit the app - as discussed on review.

File: asab/log.py [hunk 4]
```diff
@@ -451,8 +451,14 @@ def _on_write(self):
...
-				print("Error when writing to syslog '{}'".format(self._address), e, file=sys.stderr)
-				self._enqueue(msg)
+				# Contingency dump when the socket is not ready
+				print(msg.decode("utf-8"), file=sys.stderr)
+				print(
+					"Error when writing to syslog '{}': {}".format(self._address, e),
+					traceback.format_exc(),
+					sep="\n",
+					file=sys.stderr
+				)
```
Comment (byewokko) [inline asab/log.py:461]: @ateska what should be printed to stderr in this case of socket failure?
- the log message that could not be logged to socket?
- `"Error when writing to syslog ..."` message with details?
- error traceback?

**Review by ateska — APPROVED** [review]

**Pull request merged**